server/handlers/attachments.go

```diff
@@ -220,6 +222,23 @@ func (ah *ActivityHandler) handleAttachments(channelID, userID, text string, msg
 	return newText, attachments, parentID, errorFound
 }
 
+func (ah *ActivityHandler) getFileInfo(name string, fileIDs []string, fileInfos map[string]*model.FileInfo) string {
```

Comment:
To confirm, are we trying to see if an attachment is already present either in existing attachements or file store? I am a bit confused with the function here. (Sorry, it's my first time looking into this code)

Reply:
@isacikgoz - Yeah...that was a complicated and hard to understand attempt at limiting db calls to absolute necessity. I rewrote this logic which is -
- if there are attachments, retrieve the existing attachment's file Infos. 
- this is because, from MS Teams we receive a "filename", but we have fileID.
- if an attachment is already present, just use that FileID. Otherwise, download...store, create, etc.

I think the new logic is way more readable.